### PR TITLE
city population

### PR DESCRIPTION
**Pull request by bkolad**

### CODE REVIEW

File: src/main/java/net/consensys/wittgenstein/core/NetworkLatency.java [hunk 1]
```diff
@@ -143,6 +143,7 @@ public int getExtendedLatency(Node from, Node to, int delta) {
       Integer reg1 = regionPerCity.get(from.cityName);
       Integer reg2 = regionPerCity.get(to.cityName);
 
+
```
Comment (nkeywal) [inline src/main/java/net/consensys/wittgenstein/core/NetworkLatency.java]: Not a big deal but not really needed either :-)

File: src/main/java/net/consensys/wittgenstein/protocols/GSFSignature.java [hunk 2]
```diff
@@ -656,8 +656,8 @@ private static GSFSignature newProtocol() {
     double deadR = 0.10;
     double tsR = .85;
 
-    String nb = RegistryNodeBuilders.name(true, false, 0.33);
-    String nl = NetworkLatency.AwsRegionNetworkLatency.class.getSimpleName();
+    String nb = RegistryNodeBuilders.ALL_CITIES;// name(true, false, 0.33);
```
Comment (nkeywal) [inline src/main/java/net/consensys/wittgenstein/protocols/GSFSignature.java]: Why this change?

File: src/test/java/net/consensys/wittgenstein/core/CityPopulationTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    citiesInfo = cityInfoMap(map, totalPopulation);
+  }
+
+  public Map<String, CityInfo> citiesPosition() {
+    return citiesInfo;
+  }
+}
+
+
+public class CityPopulationTest {
```
Comment (nkeywal) [inline src/test/java/net/consensys/wittgenstein/core/CityPopulationTest.java]: Wow, a test :-)

File: src/test/java/net/consensys/wittgenstein/core/CityPopulationTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        new NodeBuilder.NodeBuilderWithCity(new ArrayList<>(cities.keySet()), geo);
+
+    int bigCityCount = 0;
+    for (int i = 0; i < total; i++) {
+      String city = nb.getCityName(i);
+      if (city.equals("BigCity")) {
+        bigCityCount++;
+      }
+    }
+    Assert.assertTrue("Bigger cities should be selected more often than smaller",
```
Comment (nkeywal) [inline src/test/java/net/consensys/wittgenstein/core/CityPopulationTest.java]: I'm surprised it works with just exactly 10 runs. Wouldn't it safer to have a total of 100 and check that bigCity is selected as least 40 times.

File: src/main/java/net/consensys/wittgenstein/core/NetworkLatency.java (hunk 1 above)
Comment (bkolad) [reply src/main/java/net/consensys/wittgenstein/core/NetworkLatency.java]: I will fix it

File: src/test/java/net/consensys/wittgenstein/core/CityPopulationTest.java (hunk 4 above)
Comment (bkolad) [reply src/test/java/net/consensys/wittgenstein/core/CityPopulationTest.java]: It works becouse I don't use random number to get the city but counter from the loop, so all possibilities are checked

File: src/main/java/net/consensys/wittgenstein/protocols/GSFSignature.java (hunk 2 above)
Comment (bkolad) [reply src/main/java/net/consensys/wittgenstein/protocols/GSFSignature.java]: the
> RegistryNodeBuilders.name

function currently doesn't support (All_CITIES, Real Latency) option. We can update it as a separate PR.
